### DISCUSSION TITLE
Converting double to integer using fint() is not working. Is there any alternative for this?

### DISCUSSION THREAD

**Discussion by NitishNY** [UPPAALModelChecker/UPPAAL-Meta#147] [Q&A]
I want to convert 1.8 to 2.0 and then change the type of the result to an integer value i.e., 2. To do this, I execute the following function for the shown template.

```
int xx;
double dx=round(1.8);
void cast() { xx = fint(dx); }
```
![cast](https://user-images.githubusercontent.com/113758247/190958886-f1853b56-1f4a-4dc5-86f4-af9e5451611b.png)

Upon execution, the value of xx is still 0. Could someone please help me figure this out.

**Comment by thorulf4** [comment]
Im unable to reproduce this issue. What version of uppaal are you running? And how are you testing your model, through the symbolic simulator, concrete simulator, verifier or something else?

**Comment by NitishNY** [comment]
I am using  *UPPAAL 4.1.26-1 (rev. 7BCF30B7363A9518). It supports SMC*

On Mon, Sep 19, 2022 at 12:14 PM Thorulf Neustrup ***@***.***>
wrote:

> Im unable to reproduce this issue. What version of uppaal are you running?
> And how are you testing your model, through the symbolic simulator,
> concrete simulator, verifier or something else?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/UPPAALModelChecker/UPPAAL-Meta/discussions/147#discussioncomment-3677950>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A3D5AJZOV5PMFWXLBWOBYI3V7ADVPANCNFSM6AAAAAAQPZMJPI>
> .
> You are receiving this because you authored the thread.Message ID:
> <UPPAALModelChecker/UPPAAL-Meta/repo-discussions/147/comments/3677950@
> github.com>
>

-- 
The information contained in this electronic communication is intended 
solely for the individual(s) or entity to which it is addressed. It may 
contain proprietary, confidential and/or legally privileged information. 
Any review, retransmission, dissemination, printing, copying or other use 
of, or taking any action in reliance on the contents of this information by 
person(s) or entities other than the intended recipient is strictly 
prohibited and may be unlawful. If you have received this communication in 
error, please notify us by responding to this email or telephone and 
immediately and permanently delete all copies of this message and any 
attachments from your system(s). The contents of this message do not 
necessarily represent the views or policies of BITS Pilani.

**Comment by thorulf4** [comment]
The provided example seems to work fine when using SMC features. However, doubles are not supported for non SMC features like the Simulator, Concrete Simulator and non SMC verifier queries.

![image](https://user-images.githubusercontent.com/10059450/190966798-285cc2b7-ed84-4549-85e3-07f61a9dbda4.png)

If you can provide a more clear example of how you are running into your issue I might be able to give more help.
